statistics/mode.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns the arithmetic mode(s) of the numbers in the given collection
+ *
+ * Example:
+ *
+ * ```ts
+ * import { assertEquals } from "../testing/asserts.ts"
+ * import { mode } from "./mode.ts"
+ *
+ * const numbers = [ 4, 2, 7, 4 ]
+ * assertEquals(mode(numbers), 4)
+ * ```
+ */
+export function mode(collection: Array<number>): Set<number> | undefined {
+  if (collection.length === 0) {
+    return undefined;
+  }
+
+  const counter: Record<number, number> = {};
```

Comment:
You should use `Map<number, number>`. A record will stringify the numbers in the keys technically.

```ts
> Object.entries({ 0: 0, 1: 1 })
[ [ "0", 0 ], [ "1", 1 ] ]
```

vs
```ts
> let m = new Map<number, number>()
undefined
> m.set(0, 0)
Map { 0 => 0 }
> m.set(1, 1)
Map { 0 => 0, 1 => 1 }
> [...m]
[ [ 0, 0 ], [ 1, 1 ] ]
```